File: SurvivorBot v1/src/main/java/com/fireblazerrr/survivorbot/channel/ChannelStorage.java
package com.fireblazerrr.survivorbot.channel;

import java.util.Set;

public interface ChannelStorage {

    void addChannel(Channel channel);

    void flagUpdate(Channel channel);

    Channel load(String string);

    Set<Channel> loadChannels();

    void removeChannel(Channel channel);

    void update();

    void update(Channel channel);
}

File: src/main/kotlin/com/fireblazerrr/survivorbot/channel/ChannelStorage.kt
package com.fireblazerrr.survivorbot.channel

interface ChannelStorage{

    fun addChannel(channel: Channel): Nothing
    fun flagUpdate(channel: Channel): Nothing
    fun load(string: String): Channel
    fun loadChannels(): Set<Channel>
    fun removeChannel(channel: Channel): Nothing
    fun update(): Nothing
    fun update(channel: Channel): Nothing
}